core/runtime/src/main/java/io/quarkus/runtime/logging/AsyncConfig.java

```diff
@@ -22,6 +23,7 @@
     /**
      * Determine whether to block the publisher (rather than drop the message) when the queue is full
      */
+    @DefaultConverter
```

Comment:
It's one of those breaking changes we talked about. At this stage some users are already using `BLOCK`.

Reply:
If we hyphenate the enum constants, and we hyphenate the user input, then they could continue to have `BLOCK` as a value without breaking compatibility.

Reply:
> It's one of those breaking changes we talked about. At this stage some users are already using `BLOCK`.

We have a fallback `Enum.valueOf(type, value)` option in `HyphenatedEnumConverter`. So I think this could work too.

Reply:
In that case we'd only support the fully hyphenated version (`block`) or the exact original enum constant (`BLOCK`).  Hyphenating both would allow supporting any value that maps to the hyphenated version (such as `Block`), which is slightly more flexible.

Reply:
Good remark on flexibility. I'll update this and the corresponding tests too.